awx/ui_next/src/screens/Inventory/InventoryList/useWsInventories.js

```diff
@@ -0,0 +1,92 @@
+import { useState, useEffect } from 'react';
+import useWebsocket from '../../../util/useWebsocket';
+import useThrottle from '../../../util/useThrottle';
+
+export default function useWsProjects(
+  initialInventories,
+  fetchInventoriesById
+) {
+  const [inventories, setInventories] = useState(initialInventories);
+  const [inventoriesToFetch, setInventoriesToFetch] = useState([]);
+  const throttledInventoriesToFetch = useThrottle(inventoriesToFetch, 5000);
+  const lastMessage = useWebsocket({
+    inventories: ['status_changed'],
+    jobs: ['status_changed'],
+    control: ['limit_reached_1'],
+  });
+
+  useEffect(() => {
+    setInventories(initialInventories);
+  }, [initialInventories]);
+
+  const enqueueId = id => {
+    if (!inventoriesToFetch.includes(id)) {
+      setInventoriesToFetch(ids => ids.concat(id));
+    }
+  };
+  useEffect(
+    function fetchUpdatedInventories() {
+      (async () => {
+        if (!throttledInventoriesToFetch.length) {
+          return;
+        }
+        setInventoriesToFetch([]);
+        const newInventories = await fetchInventoriesById(
+          throttledInventoriesToFetch
+        );
+        let updated = inventories;
+        newInventories.forEach(inventory => {
+          const index = inventories.findIndex(i => i.id === inventory.id);
+          if (index === -1) {
+            return;
+          }
+          updated = [
+            ...updated.slice(0, index),
+            inventory,
+            ...updated.slice(index + 1),
+          ];
```

Comment:
```suggestion
          updated[index] = inventory;
```

Reply:
That would mutate the state value directly rather than via setState, which can lead to weird bugs in React... but I think if a initialize updated with `const updated = [...inventories]` it should work.  👍🏻

Reply:
:+1: Yep - good to copy the values first. thx